pkg/rancher-desktop/config/settings.ts

```diff
@@ -186,14 +192,35 @@ export async function clear() {
 export function getUpdatableNode(cfg: Settings, fqFieldAccessor: string): [Record<string, any>, string] | null {
   const optionParts = fqFieldAccessor.split('.');
   const finalOptionPart = optionParts.pop() ?? '';
-  let currentConfig: Record<string, any> = cfg;
+  const currentConfig = optionParts.length === 0 ? cfg : _.get(cfg, optionParts.join('.'));
+
+  return (finalOptionPart in (currentConfig || {})) ? [currentConfig, finalOptionPart] : null;
+}
+
+export function getObjectRepresentation(fqFieldAccessor: string, finalValue: boolean|number|string): RecursivePartial<Settings> {
```

Comment:
The reason why you don't see this method being used outside tests is because that won't happen until the subsequent PR. I guess it could be a separate PR on that basis...